### DISCUSSION TITLE
Storylines Schema Synchronization

### DISCUSSION THREAD

**Discussion by yileifeng** [ramp4-pcar4/storylines-editor#490] [General]
We have our Storylines schema file located in the Storylines repository, which is updated regularly with new client feature requests that require additional config adjustments. There is also a copy of the Storylines schema file contained in RESPECT, used for validation in the advanced editor for slides. Originally, the idea was just to update the schema file in the editor codebase every new Storylines publish to keep everything up-to-date.

In light of recent PRs made to RESPECT, there have been editor exclusive config properties added to the Storylines schema. The main culprit recently has been adding the `modified` property to detect if panels are different from their base panel configuration, being added to every single panel definition in the schema. There is a high chance there are more instances of this contained in the editor (mainly thinking about temporary properties added to specific panels prior to saving) and would break the schema validation if left unattended. This boils down to the main concern: which is that changing the Storylines schema from the editor repo will clash with the changes made in the main Storylines repo, creating a problematic mismatch between schemas.

On a slightly different point, there is an additional `StorylinesSlideSchema.json` file in the editor codebase, which is what is being used for schema validation. This is in addition to the full `StorylinesSchema.json` file, implying there are two places to update new changes so this is also something that should be addressed as well. 

Haven't researched on potential solutions to the problem yet, so feel free to leave any suggestions in the comments.

**Comment by yileifeng** [comment]
If we determine the editor should be restricted from directly modifying the Storylines schema, this may provide the clearest path forward. We would then only keep the `StorylinesSlideSchema.json` file in the editor (used for validation and can contain additional properties required for the editor) and remove the `StorylinesSchema.json` file, eliminating the need for any schema mismatch across the two repos. 

Since the editor formulates the final configuration files on save, an important step here would be to eliminate the editor exclusive properties from being included in the output config. Otherwise, this would violate the Storylines schema as it will be unaware of editor flags used. 

In summary, for this solution: 
1. Remove `StorylinesSchema.json` file in the editor codebase (verify it is unused)
2. Ensure any editor specific flags (`modified` and other temporary panel editor properties) are not included in the config when saved (check the `saveChanges` function of each panel component)
